pkg/ssh/connection_golang.go

```diff
@@ -324,3 +356,20 @@ func getUDS(uri *url.URL, iden string) (string, error) {
 	}
 	return info.Host.RemoteSocket.Path, nil
 }
+
+// writeKnownHosts writes to the given users' known_hosts file
+func writeKnownHosts(host string, pubKey ssh.PublicKey) (string, error) {
+	hd := homedir.Get()
+	known := filepath.Join(hd, ".ssh", "known_hosts")
+	f, err := os.OpenFile(known, os.O_APPEND|os.O_WRONLY, 0o600)
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+	// normalize the line, then add it to known_hosts
+	l := knownhosts.Line([]string{host}, pubKey)
+	if _, err = f.WriteString("\n" + l); err != nil {
```

Comment:
At least, the added data should end with a newline so that things like `cat ~/.ssh/known_hosts` look right.

Non-blocking: _Typically_, the previous line already contains a newline, so adding it before the new line might not be necessary — but doing that correctly would require reading the previous contents first.

Reply:
I added the first /n as a catch all, I will put in the following one as well